bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/KeyValueStorageRocksDB.java

```diff
@@ -540,6 +551,7 @@ public void put(byte[] key, byte[] value) throws IOException {
         public void remove(byte[] key) throws IOException {
             try {
                 writeBatch.delete(key);
+                countBatchAndFlushIfNeeded();
```

Comment:
When we try to remove a key from the writeBatch, but the writeBatch has been triggered flushed in `put` method, will it impact the correctness of the key value stored in the RocksDB?

Reply:
It won't. The batch should only affect when the action starts to do in the rocksdb, we're just pending our operations into the batch to avoid calling the real action method multiple times. 
See here: https://github.com/facebook/rocksdb/blob/9a034801cead6421bcf82b506b77e3b2251f1edb/include/rocksdb/write_batch.h#L9